src/libs/elektra/kdb.c

```diff
@@ -999,6 +999,29 @@ KDB * kdbOpen (const KeySet * contract, Key * errorKey)
 		goto error;
 	}
 
+	// TODO (atmaxinger): improve
```

Comment:
I like the idea of blocking errors this way but I wouldn't call it "dummy" value. Please document the blocking in doc/METADATA.ini. Maybe it should even be its own error category, which is not allowed to be propagated to the user.

Reply:
> Maybe it should even be its own error category, which is not allowed to be propagated to the user.

I'm not so sure about this. Making this an official solution is a bit awkward. Just saying "is not allowed to be" doesn't really ensure anything. If it does end up being propagated to the user, you'll probably get very weird output, since none of the `meta:/error/*` keys is set. For example `kdb` would print (with verbose and debug output)

```
Sorry, module  issued the error :
: 
Mountpoint: 
Configfile: 
At: :

```

> Please document the blocking in doc/METADATA.ini.

Especially this part I'm not sure about.

To go by the terminology from #4436. Blocking an error like this should only be documented in "Contributor documentation" and nowhere else. It is an internal hacky solution, not something that should be encouraged.

If we want an official solution, we should create a "only warnings" error category with all the various `meta:/error/*` values. But if you use that to block errors, you'll need to use a `ksDel (ksCut (...))` to clear the block instead of a simple `keySetMeta`. So I'd think that's the best solution either.

Reply:
> If we want an official solution, we should create a "only warnings" error category with all the various meta:/error/* values.

Yes, good proposal. Probably we give it the number C01311 as it is an Internal error if the error would actually surface to the user?

IMHO such a feature is also relevant for, e.g. plugins, if they want to make sure that they can set an error later of a category they don't know yet.

Within kdb.c I already thought several times it would be good to have a way to say that all errors in a function should be emitted as a warning.

Reply:
I've created issue #4475 to disucss this further.